### ISSUE TITLE
train from scratch on ucf101 dataset

### ISSUE THREAD

**Issue by leviswind** [deepmind/kinetics-i3d#46]
We try to train i3d model on ucf101 from scratch, but it converges much slower with a final validation accuray around 60%. Can you offer some suggestions on train i3d model without imagenet pretrained model.

**Comment by joaoluiscarreira** [comment]
Hi,

i think 50-60% accuracy is to be expected when training I3D from scratch on
RGB in UCF101. If you do the same on flow it should get ~80%.
When averaging both we got 88% in the last version of the quo vadis paper.

In summary, i think your training setup should be fine.

Best,

Joao

**Comment by leviswind** [comment]
How to train i3d with optical-flow with imagenet pretrained model? Can you offer some details of training on UCF101 dataset.

**Comment by leviswind** [comment]
Also, what's the convergence speed  should be when training optical-flow compared with rgb with imagenet pretraining

**Comment by joaoluiscarreira** [comment]
The way we did it was that we inflated the weights of the imagenet model
into 3D, then trained the model normally from there, without freezing batch
norm. I think you can find code online for training the model if you search
on google. I tend to remember that the flow model converges faster but this
was a long time ago.

Best,

Joao

**Comment by leviswind** [comment]
However, the input channel of the first conv layer is 2 for flow data compared with 3 for rgb. How to deal with the difference? @joaoluiscarreira . I'm really appreciated for your help.

**Comment by joaoluiscarreira** [comment]
Actually i went back to check what exactly we did and for that particular
layer we average the original weights for the 3 input channels then copied
it twice -- so the initial weights are the same for both flow input
dimensions. But i think it did not make much of a difference compared to
the other option.

Joao

**Comment by joaoluiscarreira** [comment]
As mentioned earlier in the thread, training from scratch on flow got close
to 80%. You could try testing with batch statistics to see if there's some
batch norm moving average problem.

**Comment by poweryin** [comment]
@leviswind,Hi，can you train the i3d on ucf101 successfully? I want to use i3d on ucf101, How can I use i3d model to fine-tune on ucf101? where is the  train code?
can you give me some advice.
Thanks
Best wishes
